[CMake] Make WebKitLegacy a framework

donny-dont opened the pull request:
#### 7c4ff08c02d30dc38f4b05a58d187b2476cb0eb7
<pre>
[CMake] Make WebKitLegacy a framework
<a href="https://bugs.webkit.org/show_bug.cgi?id=249273">https://bugs.webkit.org/show_bug.cgi?id=249273</a>

Reviewed by Michael Catanzaro.

Create a framework target for WebKitLegacy. Copy the headers using
`WEBKIT_COPY_FILES`. Use the new `WebKit::WebKitLegacy` target
throughout the build by using it in `_FRAMEWORKS`.

Remove `WEBKIT_MAKE_FORWARDING_HEADERS` as it is no longer used.

* Source/WebKitLegacy/CMakeLists.txt:
* Source/WebKitLegacy/PlatformMac.cmake:
* Source/WebKitLegacy/PlatformWin.cmake:
* Source/cmake/WebKitMacros.cmake:
* Tools/DumpRenderTree/CMakeLists.txt:
* Tools/DumpRenderTree/PlatformWin.cmake:
* Tools/MiniBrowser/win/CMakeLists.txt:
* Tools/TestRunnerShared/CMakeLists.txt:
* Tools/TestWebKitAPI/CMakeLists.txt:

Canonical link: <a href="https://commits.webkit.org/257882@main">https://commits.webkit.org/257882@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7cc7bd75069f2ac61bab7778c9a73d0deda42bf5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100250 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9418 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33326 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/109574 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/169806 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/104242 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10320 "Built successfully") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/92668 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107461 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106026 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/7810 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91084 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34485 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/89730 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/22474 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/90818 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3170 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23993 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/86801 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/613 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3151 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Ventura-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29077 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9278 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43482 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/89681 "Built successfully") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5417 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/4980 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/20049 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
Can't wait to see what this breaks!

donny-dont committed:
[CMake] Make WebKitLegacy a framework
https://bugs.webkit.org/show_bug.cgi?id=249273

Reviewed by Michael Catanzaro.

Create a framework target for WebKitLegacy. Copy the headers using
`WEBKIT_COPY_FILES`. Use the new `WebKit::WebKitLegacy` target
throughout the build by using it in `_FRAMEWORKS`.

Remove `WEBKIT_MAKE_FORWARDING_HEADERS` as it is no longer used.

* Source/WebKitLegacy/CMakeLists.txt:
* Source/WebKitLegacy/PlatformMac.cmake:
* Source/WebKitLegacy/PlatformWin.cmake:
* Source/cmake/WebKitMacros.cmake:
* Tools/DumpRenderTree/CMakeLists.txt:
* Tools/DumpRenderTree/PlatformWin.cmake:
* Tools/MiniBrowser/win/CMakeLists.txt:
* Tools/TestRunnerShared/CMakeLists.txt:
* Tools/TestWebKitAPI/CMakeLists.txt:

Canonical link: https://commits.webkit.org/257882@main
